main.go

```diff
@@ -47,10 +49,67 @@ func bod(t time.Time) time.Time {
 func parse_d_tag(tag string) (t time.Time, err error) {
 	const shortForm = "d_2006_01_02"
 
+	parts := strings.Split(tag, "_")
+	if len(parts) != 4{
+		return time.Time{}, fmt.Errorf("Got invalid format")
+	}
+	if strings.ToLower(parts[0]) != "d" {
+		return time.Time{}, fmt.Errorf("Got the wrong tag, should be a daily, got %s",parts[0])
+	}
+	month, err := strconv.Atoi((parts[2]))
+	if err != nil{
+		return time.Time{}, err
+	}
+	day, err := strconv.Atoi((parts[3]))
+	if err != nil{
+		return time.Time{}, err
+	}
+	if len(parts[1]) != 4 {
+		return time.Time{}, fmt.Errorf("Got the wrong year format, got %s",parts[1])
+	}
+	if len(parts[2]) != 2 || month >12 || month<01{
+		return time.Time{}, fmt.Errorf("Got the wrong month format, got %s",parts[2])
+	}
+	if len(parts[3]) != 2 || day >31 || day <01{
+		return time.Time{}, fmt.Errorf("Got the wrong day format, got %s",parts[2])
+	}
+
 	t, err = time.Parse(shortForm, tag)
 	return t, err
 }
 
+func parse_w_tag(tag string) (t time.Time, err error) {
```

Comment:
This always returns a Monday with the exception of the first release in a year which is always Jan 1st.
Check for consistency.